application/src/main/java/org/thingsboard/server/service/entitiy/alarm/DefaultTbAlarmService.java

```diff
@@ -221,36 +225,50 @@ public AlarmInfo unassign(Alarm alarm, long unassignTs, User user) throws Things
+    public List<AlarmId> unassignDeletedUserAlarms(TenantId tenantId, User user, long unassignTs) {
+        List<AlarmId> totalAlarmIds = new ArrayList<>();
+        List<AlarmId> alarmIds;
+        do {
+            alarmIds = alarmService.findAlarmIdsByAssigneeId(tenantId, user.getId(), 100);
+            for (AlarmId alarmId : alarmIds) {
+                log.trace("[{}] Unassigning alarm {} userId {}", tenantId, alarmId.getId(), user.getId().getId());
+                AlarmApiCallResult result = alarmSubscriptionService.unassignAlarm(user.getTenantId(), alarmId, unassignTs);
+                Alarm alarm = result.getAlarm();
                 if (!result.isSuccessful()) {
                     continue;
                 }
                 if (result.isModified()) {
-                    AlarmComment alarmComment = AlarmComment.builder()
-                            .alarmId(alarm.getId())
-                            .type(AlarmCommentType.SYSTEM)
-                            .comment(JacksonUtil.newObjectNode().put("text", String.format("Alarm was unassigned because user %s - was deleted",
-                                            (user.getFirstName() == null || user.getLastName() == null) ? user.getName() : user.getFirstName() + " " + user.getLastName()))
-                                    .put("userId", user.getId().toString())
-                                    .put("subtype", "ASSIGN"))
-                            .build();
                     try {
-                        alarmCommentService.saveAlarmComment(alarm, alarmComment, user);
+                        AlarmComment alarmComment = AlarmComment.builder()
+                                .alarmId(alarmId)
+                                .type(AlarmCommentType.SYSTEM)
+                                .comment(JacksonUtil.newObjectNode()
+                                        .put("text", String.format("Alarm was unassigned because user with id %s - was deleted",
+                                                (user.getFirstName() == null || user.getLastName() == null) ? user.getName() : user.getFirstName() + " " + user.getLastName()))
+                                        .put("userId", user.getId().toString())
+                                        .put("subtype", "ASSIGN"))
+                                .build();
+                        alarmCommentService.saveAlarmComment(alarm, alarmComment, null);
                     } catch (ThingsboardException e) {
                         log.error("Failed to save alarm comment", e);
                     }
                     notificationEntityService.logEntityAction(alarm.getTenantId(), alarm.getOriginator(), result.getAlarm(),
-                            alarm.getCustomerId(), ActionType.ALARM_UNASSIGNED, user);
+                            alarm.getCustomerId(), ActionType.ALARM_UNASSIGNED, null);
```

Comment:
We have such a detail as a user. Notification plays a role as an audit log as well. 
Why do we avoid users in the audit log?